src/diffusers/pipelines/pipeline_loading_utils.py

```diff
@@ -184,12 +199,24 @@ def variant_compatible_siblings(filenames, variant=None) -> Union[List[os.PathLi
 
     # all variant filenames will be used by default
     usable_filenames = set(variant_filenames)
+    transformers_weight_prefixes = ("pytorch_model", "model")
+
+    def is_transformers_file(filename):
+        return filename.split("/")[-1].startswith(transformers_weight_prefixes)
 
     def convert_to_variant(filename):
         if "index" in filename:
-            variant_filename = filename.replace("index", f"index.{variant}")
+            if is_transformers_file(filename):
+                variant_filename = filename.replace("index", f"index.{variant}")
+            else:
+                variant_filename = filename.replace("index", f"{variant}.index")
```

Comment:
Because:

```
# For transformers, it will be `text_encoder/pytorch_model.bin.index.fp16.json`
# for diffusers, it will be `unet/diffusion_pytorch_model.safetensors.fp16.index.json`
```